packages/admin/admin/src/appHeader/button/AppHeaderButton.tsx

```diff
@@ -1,64 +1,62 @@
+import { AppHeaderButtonClassKey, EndIcon, Inner, Root, StartIcon, Text } from "./AppHeaderButton.styles";
+
+export interface AppHeaderButtonProps
+    extends ButtonBaseProps,
+        ThemedComponentBaseProps<{
+            root: typeof ButtonBase;
+            inner: "div";
+            text: typeof Typography;
+            startIcon: "div";
+            endIcon: "div";
+        }> {
     startIcon?: React.ReactNode;
     endIcon?: React.ReactNode;
     disableTypography?: boolean;
 }
 
-function Button({
-    classes,
-    children,
-    startIcon,
-    endIcon,
-    disableTypography,
-    onClick,
-    ...restProps
-}: AppHeaderButtonProps & WithStyles<typeof styles>): React.ReactElement {
-    const {
-        startIcon: startIconClassName,
-        endIcon: endIconClassName,
-        typography: typographyClassName,
-        inner: innerClassName,
-        ...buttonBaseClasses
-    } = classes;
+export function AppHeaderButton(inProps: AppHeaderButtonProps) {
+    const { children, disableTypography, slotProps, onClick, startIcon, endIcon, ...restProps } = useThemeProps({
+        props: inProps,
+        name: "CometAdminAppHeaderButton",
+    });
 
     return (
-        <ButtonBase classes={buttonBaseClasses} {...restProps} onClick={onClick}>
-            <div className={innerClassName}>
-                {startIcon && <div className={startIconClassName}>{startIcon}</div>}
+        <Root {...restProps} {...slotProps?.root} onClick={onClick}>
+            <Inner {...slotProps?.inner}>
+                {startIcon && <StartIcon {...slotProps?.startIcon}>{startIcon}</StartIcon>}
                 {children &&
                     (disableTypography ? (
-                        children
+                        { children }
                     ) : (
-                        <Typography component="div" classes={{ root: typographyClassName }}>
+                        // @ts-expect-error TODO
+                        <Text component="div" {...slotProps?.text}>
                             {children}
-                        </Typography>
+                        </Text>
                     ))}
-                {endIcon && <div className={endIconClassName}>{endIcon}</div>}
-            </div>
-        </ButtonBase>
+
+                {endIcon && <EndIcon {...slotProps?.endIcon}>{endIcon}</EndIcon>}
+            </Inner>
+        </Root>
     );
 }
 
-export const AppHeaderButton = withStyles(styles, { name: "CometAdminAppHeaderButton" })(Button);
-
 declare module "@mui/material/styles" {
-    interface ComponentNameToClassKey {
-        CometAdminAppHeaderButton: AppHeaderButtonClassKey;
+    interface ComponentsPropsList {
+        CometAdminAppHeaderButton: AppHeaderButtonProps;
     }
 
-    interface ComponentsPropsList {
-        CometAdminAppHeaderButton: Partial<AppHeaderButtonProps>;
+    interface ComponentNameToClassKey {
+        CometAdminAppHeaderButton: AppHeaderButtonClassKey;
```

Comment:
Why did you switch those two interfaces?

Reply:
I just ordered them like they are shown in the guide. Is there a problem with reordering them?

Reply:
No, it just complicates review 🙂